images/algod/setup.py

```diff
@@ -102,11 +104,19 @@ def configure_data_dir(network_dir, token, algod_port, kmd_port, bootstrap_url):
         f.write(token)
 
     # Setup config, inject port
-    with open(join(node_dir, 'config.json'), 'w') as f:
-        f.write('{ "Version": 12, "GossipFanout": 1, "EndpointAddress": "0.0.0.0:%s", "DNSBootstrapID": "%s", "IncomingConnectionsLimit": 0, "Archival":false, "isIndexerActive":false, "EnableDeveloperAPI":true}' % (algod_port, bootstrap_url))
-    with open(join(kmd_dir, 'kmd_config.json'), 'w') as f:
-        f.write('{  "address":"0.0.0.0:%s",  "allowed_origins":["*"]}' % kmd_port)
-
+    node_config_path = join(node_dir, "config.json")
+    archival = 'true' if archival else 'false'
+    is_indexer_active = 'true' if is_indexer_active else 'false'
+    node_config = f'{{ "Version": 12, "GossipFanout": 1, "EndpointAddress": "0.0.0.0:{algod_port}", "DNSBootstrapID": "{bootstrap_url}", "IncomingConnectionsLimit": 0, "Archival":{archival}, "isIndexerActive":{is_indexer_active}, "EnableDeveloperAPI":true }}'    
```

Comment:
nice f-str - is there a guarantee that `setup.py` will always run on python >=3.6?

Reply:
We get to control that in Algod's Dockerfile. Though we're not explicitly ensuring python >= 3.6, I'm not the first to introduce f-strings here. (e.g. see [this PR from July 2020](https://github.com/algorand/sandbox/blob/d70ffd7c589b699cf275402ba187005097abdf02/images/algod/setup.py#L143))